### ISSUE TITLE
Nerozpoznaný typ digitálního objektu (DigitalObject) supplement

### ISSUE THREAD

**Issue by daliboris** [LIBCAS/DL4DH#63]
### Postup

- publikace s `uuid:587dec55-82d2-4c9b-a45c-da482e56d8e8`
- obohacení publikace (`/api/enrich/download-k-structure`)

### Výsledek

`DOWNLOAD_K_STRUCTURE` `FAILED`

### Log
JSON decoding error: Could not resolve type id 'supplement' as a subtype of
 `cz.inqool.dl4dh.krameriusplus.core.system.digitalobject.DigitalObject`: 
known type ids = [internalpart, monograph, monographunit, page, periodical, periodicalitem, periodicalvolume]; 
nested exception is com.fasterxml.jackson.databind.exc.InvalidTypeIdException: 
Could not resolve type id 'supplement' as a subtype of 
`cz.inqool.dl4dh.krameriusplus.core.system.digitalobject.DigitalObject`: 
known type ids = [internalpart, monograph, monographunit, page, periodical, periodicalitem, periodicalvolume] 
at [Source: (io.netty.buffer.ByteBufInputStream); line: 1, column: 4330]

**Comment by bodnarIQ** [comment]
Korektnym spravanim by malo byt v pripade neocakavaneho typu digitalneho objektu takyto typ ignorovat a pokracovat v spracovavani?

**Comment by daliboris** [comment]
Aplikace by měla u neznámého typu digitálního objektu generovat chybu jako doposud. Z analýzy pak vznikne požadavek na úpravu aplikace: buď aby si s novým typem poradila, nebo ho ignorovala.

V tomto konkrétním případě je potřeba opravit aplikaci, aby typ dokumentu `supplement` dokázala zpracovat; jde o přílohu, což je u novin a časopisů obvyklé.

**Comment by daliboris** [comment]
Ano, přesně tak, z pohledu koncového uživatele/badatele je to typ publikace, která má strany. Liší se bibliografickým popisem a vztahem k nadřazeným jednotkám. (To jsou ale detaily, do kterých vidí spíše @zabak).

**Comment by daliboris** [comment]
Po otestování končí úloha obohacení publikace chybou `
403 Forbidden from GET https://www.ndk.cz/search/api/v5.0/item/uuid:84aa0a30-e7b2-11e6-b71a-005056822549/streams/ALTO`.

Jde o to, že tato publikace patří k DNNT (dílo nedostupné na trhu), takže text není z licenčních důvodů pro uživatele dostupný.

Je potřeba vyřešit přístup k DNNT, popř. dalším dílům, které jsou chráněny AZ.
